Add VideoPress "benefit" card to plan removal screen

jgcaruso opened the pull request:
#### Proposed Changes

* Add details about how many videos will be not accessible when removing a Jetpack plan when VideoPress is used

Jetpack VideoPress Product
<img width="1197" alt="Screen Shot 2022-08-31 at 9 47 11 AM" src="https://user-images.githubusercontent.com/4081020/187701839-400a9918-91ac-407b-a737-ee72ba584217.png">


Jetpack Complete (includes VideoPress)
<img width="1202" alt="Screen Shot 2022-08-31 at 9 46 48 AM" src="https://user-images.githubusercontent.com/4081020/187701864-c5e1376b-5964-4679-960f-b4b666d3df9a.png">



#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Pre-testing: Create a new site with a VideoPress plan and upload a few videos to it (if you don't already have one)
* visit `/me/purchases/`
* select your Jetpack site
* Click "Remove [Jetpack Plan]" button
* The modal that appears should include a Video section with the correct VideoPress video count
* Select another Jetpack site that doesn't have a plan that includes VideoPress (eg add the Security plan to a new site)
* Click "Remove plan" button -> You should **NOT** see the VideoPress "benefit" card
* Manual Regression test: Visit `/media/[some site]`, flip quickly between the filters (Images, Videos) and ensure the correct data is shown (from this issue https://github.com/Automattic/wp-calypso/pull/44579)
* Manual Regression test 2: Visit `/stats/month/[some site]`, ensure the `Videos` section has data (this is an example of `<QueryMedia>` being used without a query specified.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #66076

jgcaruso committed:
add check and benefit card for videopress -- video count is just a placeholder, need to add logic to retrieve correct value

jgcaruso committed:
change label to be less vague (and not misleading)

jgcaruso committed:
(maybe) fix for media request not being being in state

jgcaruso committed:
refactored videopress benefits card into separate file

jgcaruso committed:
use placeholder while data is loading

jgcaruso committed:
only allow early return for media library filter issue on the `media` route, so other uses of MediaQuery component will continue to work

jgcaruso committed:
cleanup comments

jgcaruso committed:
remove unused prop (copy-pasta)

jgcaruso committed:
handle nulls for current route

jgcaruso commented on client/state/data-layer/wpcom/sites/media/index.js:
```diff
@@ -59,6 +60,7 @@ export const requestMediaSuccess =
 	( { siteId, query }, data ) =>
 	( dispatch, getState ) => {
 		if (
+			getCurrentRoute( getState() )?.startsWith( '/media/' ) &&
 			! isEqual(
 				omit( query, 'page_handle' ),
 				omit( getNextPageQuery( getState(), siteId ), 'page_handle' )
```
@flootr Tagging you here since you originally make this fix for https://github.com/Automattic/wp-calypso/pull/44579

The query check was preventing the request from the `QueryMedia` react component from completing anywhere except the Media Library in calypso because the "next page query" url always had `number` and none of the passed in query filters (like `mime_type`) as properties and the custom query that I provided to the component just had my `mime_type` property. As a result it would always fall into this condition and not populate the `media` state property, and my component would never get the updated data from the request.

It doesn't look like this change I'm making is causing the original issue to come back, unless I'm not testing correctly. Just hoping to get a second set of eyes on this incase you see an issue with the change.

Alternatively, it sounds like the original issue that was being fixed has something to do with old requests not being cancelled... I wonder if there is another way to solve for that that wouldn't require a special case to skip this check for the `/media` route? Something like storing a "current query" instead and just comparing that with whatever is coming back in the result?

jgcaruso replied:
I've actually replaced this change with a bug fix. Took a bit of investigating, but it looks like the `<QueryMedia>` component wasn't setting the `query` state, which was causing the `isEqual` check in the above code to trigger when it shouldn't.

This has now been fixed directly in the `<QueryMedia>` component!

roundhill reviewed (approved):
Testing steps worked well and code LGTM.

jgcaruso committed:
replace `/media` route special case with a bugfix

jgcaruso committed:
Merge branch 'trunk' into update/videopress-add-benefits-screen

jgcaruso commented:
> Testing steps worked well and code LGTM.

Thanks! Before deploying I gave it another look and found out that one of the fixes I did was actually not necessary since the behaviour was caused by something completely different in `client/components/data/query-media/index.jsx`. I fixed that instead and added an extra last step to the test plan.

Could you give the PR another review when you have time?

jgcaruso committed:
Merge branch 'trunk' into update/videopress-add-benefits-screen

roundhill reviewed (approved):
Still works great, the new regression tests were fine.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7526400

Thank you @jgcaruso for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7526400).
